File: 프로그래머스/1/92334. 신고 결과 받기/신고 결과 받기.py
def solution(id_list, report, k):
    n=len(id_list)
    answer = [0]*n
    id_index={}
    complain_count=[0]*n
    complain_list={}
        
    for idx,id in enumerate(id_list):
        id_index[id]=idx
        complain_list[id]=[]
    
    for r in set(report):
        a,b=r.split()
        complain_list[a].append(b)
        complain_count[id_index[b]]+=1

    for idx,cnt in enumerate(complain_count):
        if cnt>=int(k):
            for key,val in complain_list.items():
                if id_list[idx] in val:
                    answer[id_index[key]]+=1
    return answer

File: 프로그래머스/1/92334. 신고 결과 받기/신고 결과 받기.java
import java.util.*;
class Solution {
    public int[] solution(String[] id_list, String[] report, int k) {
        int[] answer = new int [id_list.length];
        HashMap<String, ArrayList<String>> map = new HashMap();
        HashMap<String, Integer> complain = new HashMap();
        
        for(String id: id_list){
            map.put(id, new ArrayList<String>());
            complain.put(id, 0);
        }
        
        report = Arrays.stream(report).distinct().toArray(String[]::new);
        
        for(int i=0;i<report.length;i++){
            String [] sarr = report[i].split(" ");
            if(!map.get(sarr[0]).contains(sarr[1])){
                map.get(sarr[0]).add(sarr[1]);
                complain.put(sarr[1], complain.get(sarr[1])+1);
            }
        }
        
        for(String id: id_list){
            if(complain.get(id)>=k){
                for(int j=0;j<id_list.length;j++){
                    if(map.get(id_list[j]).indexOf(id)!=-1){
                        answer[j]++;
                    }
                }
            }
        }
        return answer;
    }
}